src/Tribe/Cache/Abstract_Cache.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Implements methods common to all caches implementations.
+ */
+abstract class Tribe__Tickets__Cache__Abstract_Cache implements Tribe__Tickets__Cache__Cache_Interface {
+	/**
+	 * @var array
+	 */
+	protected $keys = array(
+		'posts_with_tickets',
+		'posts_without_tickets',
+		'past_events',
+	);
+
+	/**
+	 * @var int The expiration time in seconds.
+	 */
+	protected $expiration = 60;
+
+	/**
+	 * Sets the expiration time for the cache.
+	 *
+	 * @param int $seconds
+	 *
+	 * @return void
+	 */
+	public function set_expiration_time( $seconds ) {
+		$this->expiration = $seconds;
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function fetch_posts_with_ticket_types() {
+		$supported_types = tribe_get_option( 'ticket-enabled-post-types', array() );
+
+		if ( empty( $supported_types ) ) {
+			$ids = array();
+		}
+
+		/** @var \wpdb $wpdb */
+		global $wpdb;
+
+		$post_types = "('" . implode( "','", $supported_types ) . "')";
```

Comment:
Should `$supported_types` be cleansed in any way (maybe using `array_map( 'esc_sql', $supported_types )`) before tossing into the query?